.github/workflows/build.yml

```diff
@@ -1,23 +1,116 @@
+        default: 'false'
+        type: choice
+        options:
+        - 'true'
+        - 'false'
+
+env:
+  REGISTRY: ghcr.io
+  OWNER: security-tools-alliance
+  PROJECT: rengine-ng
 
 jobs:
-  build:
+  build-and-push:
     runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        image: [celery, web, postgres, redis, ollama, certs, proxy]
+        platform: [linux/amd64, linux/arm64]
     steps:
-      - name: Checkout the repo
+      - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Log in to GitHub's Container Registry
-        run: echo "${{ secrets.CONTAINER_REGISTRY_SECRET }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ env.PROJECT }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=sha
+            type=raw,value=latest,enable={{is_default_branch}}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      
+      - name: Login to GHCR
+        if: github.event_name != 'pull_request' || github.event.inputs.push_image == 'true'
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ vars.GHCR_USERNAME }}
+          password: ${{ secrets.GHCR_PAT }}
+      
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          context: ./docker/${{ matrix.image }}
+          file: ./docker/${{ matrix.image }}/Dockerfile
+          push: ${{ github.event_name != 'pull_request' || github.event.inputs.push_image == 'true' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          platforms: ${{ matrix.platform }}
+          outputs: type=docker,dest=/tmp/image.tar
+
+      - name: Push image if exists
+        if: github.event.inputs.push_image == 'true'
+        run: |
+          if [ -f /tmp/image.tar ]; then
+            docker load --input /tmp/image.tar
+            docker push ${{ steps.meta.outputs.tags }}
+          else
+            echo "No image found to push"
+          fi
 
-      - name: Build the Docker image
-        run: docker build . -t ghcr.io/Security-Tools-Alliance/rengine-ng:latest
+  update-release:
+    needs: build-and-push
+    if: github.event_name == 'release' && github.event.action == 'published'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-      - name: Push the Docker image
-        run: docker push ghcr.io/Security-Tools-Alliance/rengine-ng:latest
+      - name: Update release description
```

Comment:
What does this do? Does it include the assets in the release?

Reply:
It updates the release description with the associated docker images

Reply:
I don't quite understand what you mean exactly. Do you have an example? Is it just that the changelog/release description will contain <image-names>?

Reply:
yes

Reply:
But why? For what reason?

Reply:
To have the corresponding images related to the release

Reply:
What reason, maybe someone want to know what's the image for this release

Reply:
Can you show an example?

Reply:
I could remove if you don't want
We need to have this PR pushed and go to the next one
If it don't work or bad we could modify it later